addons/account/models/account_report.py

```diff
@@ -497,6 +497,9 @@ def write(self, vals):
 
         for country, former_formulas_list in former_formulas_by_country.items():
             for former_formula in former_formulas_list:
+                #if new engine is tax_tags and formula is unchanged
+                if vals.get('engine') == 'tax_tags' and 'formula' not in vals:
+                    vals['formula'] = former_formula
                 new_tax_tags = self.env['account.account.tag']._get_tax_tags(vals['formula'], country.id)
```

Comment:
If you set `vals['formula'] = former_formula`, most of the following computation will be useless, starting with  `former_tax_tags`, which will be identical to `new_tax_tags`, and it's not that great :blush: 
Could you find a way to optimize that?